### DISCUSSION TITLE
[QUESTION] Can I update the longhorn I have on my rancher via kubectl?

### DISCUSSION THREAD

**Discussion by elton557** [longhorn/longhorn#4819] [Q&A]
Can I update the longhorn I have on my rancher via kubectl? with this script

kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/v1.3.2/deploy/longhorn.yaml

**Comment by PhanLe1010** [comment]
No, it is not recommended to mix Rancher UI with kubectl. If you installed Longhorn using Rancher UI, you should continue using Rancher UI to upgrade Longhorn

**Comment by elton557** [comment]
@PhanLe1010 
It's just that since version 2.6.5 I've never been able to update through the interface, only via kubectl

**Comment by PhanLe1010** [comment]
What is the error when you tried to upgrade using the Rancher UI?

**Comment by elton557** [comment]
@PhanLe1010 No error appears, it just doesn't show the upgrade, I don't see how to update, in version 2.5 of rancher it appeared!

**Comment by joshimoo** [comment]
@elton557 is it possible that you installed via the old catalog in rancher 2.5 and now are using rancher 2.6 which only shows the new catalog?  here is a migration guide if that is the case https://longhorn.io/kb/how-to-migrate-longhorn-chart-installed-in-old-rancher-ui-to-the-chart-in-new-rancher-ui/

**Comment by elton557** [comment]
@joshimoo 
I ran it, but it returns this:

k8s-cluster-matriz on  master [?] took 3,0s 
➜ sh -x longhorn_rancher_chart_migration.sh -u /Users/elton.gomes/Documents/REPOSITORIOS/k8s-cluster-matriz/cluster-matriz.yml -d /Users/elton.gomes/Documents/REPOSITORIOS/k8s-cluster-matriz/cluster-matriz.yml --type migrate 
+ kubectl get-all version
+ '[' 0 -ne 0 ']'
+ set -e
++ dirname longhorn_rancher_chart_migration.sh
+ SCRIPT_DIR=.
+ UPSTREAM_KUBECONFIG=
+ DOWNSTREAM_KUBECONFIG=
+ KUBECTL_DRY_RUN=
+ '[' -u '!=' '' ']'
+ case $1 in
+ shift
+ UPSTREAM_KUBECONFIG=/Users/elton.gomes/Documents/REPOSITORIOS/k8s-cluster-matriz/cluster-matriz.yml
+ shift
+ '[' -d '!=' '' ']'
+ case $1 in
+ shift
+ DOWNSTREAM_KUBECONFIG=/Users/elton.gomes/Documents/REPOSITORIOS/k8s-cluster-matriz/cluster-matriz.yml
+ shift
+ '[' --type '!=' '' ']'
+ case $1 in
+ shift
+ TYPE=migrate
+ shift
+ '[' '' '!=' '' ']'
+ '[' -z /Users/elton.gomes/Documents/REPOSITORIOS/k8s-cluster-matriz/cluster-matriz.yml ']'
+ '[' -z /Users/elton.gomes/Documents/REPOSITORIOS/k8s-cluster-matriz/cluster-matriz.yml ']'
+ '[' migrate '!=' migrate ']'
+ RELEASE_NAMESPACE=longhorn-system
+ RELEASE_NAME=longhorn-system
+ echo 'Looking up Rancher Project App '\''longhorn-system'\'' ...'
Looking up Rancher Project App 'longhorn-system' ...
++ cat /Users/elton.gomes/Documents/REPOSITORIOS/k8s-cluster-matriz/cluster-matriz.yml
++ grep 'server:.*https://.*/k8s/clusters/.*'
++ awk -F/ '{print $(NF)}'
++ awk '-F"' '{print $1}'
+ DOWNSTREAMCLUSTERID=c-d8ndc
++ kubectl --kubeconfig /Users/elton.gomes/Documents/REPOSITORIOS/k8s-cluster-matriz/cluster-matriz.yml get --all-namespaces apps.project.cattle.io -o 'jsonpath={range.items[*]}{.metadata.namespace} {.metadata.name} {.spec.targetNamespace} {.spec.projectName} {.spec.externalId}{"\n"}{end}'
++ grep -s 'longhorn-system longhorn-system c-d8ndc'
error: the server doesn't have a resource type "apps"
+ RANCHERAPP=

**Reply by PhanLe1010** [reply]
Probably you didn't install Longhorn using the old Rancher app catalog. Can you upload a support bundle to `longhorn-support-bundle@suse.com` to verify?

**Comment by elton557** [comment]
@PhanLe1010 
hello, follow the support package, I'll send it to the email too:

https://drive.google.com/file/d/1HtRudF-tOECN9qJKnDTX0Ie71bBIOFF2/view?usp=sharing

**Comment by elton557** [comment]
@PhanLe1010  did you see this message above?
